### DISCUSSION TITLE
How to get all client IDs and project ID in aud claim for JWT access token obtained with Device Authorization flow?

### DISCUSSION THREAD

**Discussion by geiravos** [zitadel/zitadel#7010] [Q&A] [answered]
In my configuration I have a project with two OIDC applications. One web application using PKCE and one native using Device Authorization flow with `JWT` auth token type.

When obtaining tokens for the web application, I get access tokens with `aud` claim containing all client IDs (both the web app and native app) and also the project ID (as expected)

However, when using the device authorization flow for the native application, only the client ID for the native application is in the `aud` claim. I want this token to also contain audience for all client IDs + the project ID.

I've also tried with using the `urn:zitadel:iam:org:project:id:{projectid}:aud` scope, but still no project ID in the `aud` claim for the Device Autorization flow

Is this a bug, or is there some extra configuration needed for this to work?

**Comment by hifabienne** [comment]
@muhlemmer @livio-a can you help here?

**Reply by muhlemmer** [reply, marked as answer]
We have an open feature issue for that already: https://github.com/zitadel/oidc/issues/425

**Reply by geiravos** [reply]
Thank you for the quick reply.

As a workaround for now, I think I'll get by with validating the `aud` using the client ID of the native application instead of the proect ID in my resource servers, since it is the lowest common denominator of the tokens I'll have flowing in my applications at the moment.
